### DISCUSSION TITLE
How to create ValidationBehavior without throwing exception and return validation errors?

### DISCUSSION THREAD

**Discussion by odysseus1973** [ImmediatePlatform/Immediate.Handlers#172] [Q&A]
How to create ValidationBehavior without throwing exception and return validation errors?

**Comment by viceroypenguin** [comment]
You would create your own, similar to the one baked in:
```cs
public sealed class ValidationBehavior<TRequest, TResponse>
	: Behavior<TRequest, TResponse>
	where TRequest : IValidationTarget<TRequest>
	where TResponse : SomeResultType<TResponse>
{
	/// <summary>
	///	    Validate the <paramref name="request"/> and throw an exception if it fails.
	/// </summary>
	/// <inheritdoc />
	/// <exception cref="ValidationException">
	///	    Thrown if the <paramref name="request"/> does not validate successfully.
	/// </exception>
	public override async ValueTask<TResponse> HandleAsync(TRequest request, CancellationToken cancellationToken)
	{
		var result = TRequest.Validate(request);
		if (!result.IsValid)
			return new SomeResultType<TResponse>(failed);

		return await Next(request, cancellationToken).ConfigureAwait(false);
	}
}
```

This is not built-in because it is not possible for IV to know all potential result types that users may use, or how to correctly interact with them. Code is provided as example; further editing will be required.

**Comment by odysseus1973** [comment]
Thanks for answer! Trying everything with Result<T> from FluentResult library but without success. In my case I need return Result with validation errors (from FluentValidation library) when validation errors exists or return Result<T> when validation success.
